pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -307,20 +314,33 @@ protected void internalDeleteNamespace(AsyncResponse asyncResponse, boolean auth
         }
 
         // remove from owned namespace map and ephemeral node from ZK
-        final List<CompletableFuture<Void>> futures = Lists.newArrayList();
+        final List<CompletableFuture<Void>> deletePartitionedTopicFuture = Lists.newArrayList();
         // remove system topics first.
-        if (!topics.isEmpty()) {
-            for (String topic : topics) {
+        if (!partitionedTopicNames.isEmpty()) {
+            for (String topic : partitionedTopicNames) {
                 try {
-                    futures.add(pulsar().getAdminClient().topics().deleteAsync(topic, true, true));
```

Comment:
You only need to distinct partitionedTopicName and nonPartitionedTopicName here.

```java
TopicName topicName = TopicName.get(topic);
if (topicName.isPartitioned()) {
    futures.add(pulsar().getAdminClient().topics().deletePartitionedTopicAsync(topic, true));
} else {
    futures.add(pulsar().getAdminClient().topics().deleteAsync(topic, true));
}
```

Reply:
This is A good suggestion, but 'topicname.get(topic name).ispartitioned()' just uses the naming rule to figure out if it's a `partitioned topic`, which isn't exact. When enabled the feature `auto create topic`, users can create a non-partitioned topic name with `persistent://public/default/special-partition-0`

Reply:
Oh, I see, if so, I suggest we need to check the topic metadata.

Reply:
yes. `getPartitionedTopicList` did it